packages/packages/top-bar/src/components/actions/toggle-action.tsx

```diff
@@ -0,0 +1,28 @@
+import { useMenuContext } from '../../contexts/menu-context';
+import PopoverMenuItem from '../ui/popover-menu-item';
+import ToolbarMenuToggleItem from '../ui/toolbar-menu-toggle-item';
+import { ElementType } from 'react';
+
+type Props = {
+	title: string;
+	icon: ElementType;
+	selected?: boolean;
+	disabled?: boolean;
+	onClick?: () => void;
+	value?: string;
+};
+
+export default function ToggleAction( { icon: Icon, title, value, ...props }: Props ) {
+	const { type } = useMenuContext();
+
+	return type === 'toolbar' ? (
```

Comment:
Consider creating a shared component for all these appearances, it repeats itself in multiple components.

Reply:
actually, I prefer a small repetition over trying to find common patterns which are not really common, each component is different, and does different things.